### ISSUE TITLE
excessive_precision suggest invalid code `inf_f32`

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9910]
### Summary

.

### Lint Name

excessive_precision

### Reproducer

I tried this code:

```rust
#![deny(overflowing_literal)]
fn main() {
    let x = -3.40282357e+38_f32; //~ ERROR literal out of range for `f32`
}
```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::excessive_precision`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0425]: cannot find value `inf_f32` in this scope
 --> src/main.rs:5:15
  |
5 |     let _x = -inf_f32; //~ ERROR literal out of range for `f32`
  |               ^^^^^^^ not found in this scope

warning: unknown lint: `overflowing_literal`
 --> src/main.rs:3:9
  |
3 | #![deny(overflowing_literal)]
  |         ^^^^^^^^^^^^^^^^^^^ help: did you mean: `overflowing_literals`
  |
  = note: `#[warn(unknown_lints)]` on by default

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0425`.
Original diagnostics will follow.

warning: unknown lint: `overflowing_literal`
 --> src/main.rs:3:9
  |
3 | #![deny(overflowing_literal)]
  |         ^^^^^^^^^^^^^^^^^^^ help: did you mean: `overflowing_literals`
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: unused variable: `x`
 --> src/main.rs:5:9
  |
5 |     let x = -3.40282357e+38_f32; //~ ERROR literal out of range for `f32`
  |         ^ help: if this is intentional, prefix it with an underscore: `_x`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: float has excessive precision
 --> src/main.rs:5:14
  |
5 |     let x = -3.40282357e+38_f32; //~ ERROR literal out of range for `f32`
  |              ^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `inf_f32`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision
  = note: requested on the command line with `-W clippy::excessive-precision`

warning: literal out of range for `f32`
 --> src/main.rs:5:14
  |
5 |     let x = -3.40282357e+38_f32; //~ ERROR literal out of range for `f32`
  |              ^^^^^^^^^^^^^^^^^^
  |
  = note: the literal `3.40282357e+38_f32` does not fit into the type `f32` and will be converted to `f32::INFINITY`
  = note: `#[warn(overflowing_literals)]` on by default

warning: `clpy` (bin "clpy") generated 4 warnings (run `cargo fix --bin "clpy"` to apply 2 suggestions)
warning: `clpy` (bin "clpy" test) generated 4 warnings (4 duplicates)
    Finished dev [unoptimized + debuginfo] target(s) in 0.39s
```

I expected to see this happen:

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
